### PR TITLE
[RN] Toolbox disabled button styles

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: react/features/base/toolbox/components/AbstractToolboxItem.js [hunk 1]
```diff
@@ -7,17 +7,17 @@ export type Styles = {
     /**
      * Style for the item's icon.
      */
-    iconStyle: Object,
+    iconStyle: Object | Array<*>,
```
Comment (zbettenbuk) [inline react/features/base/toolbox/components/AbstractToolboxItem.js]: Not sure what flow accepts, but shouldn't this be Array<Object>?

Comment (saghul) [reply react/features/base/toolbox/components/AbstractToolboxItem.js]: We want to accept either a single style or an array of them, hence the "or".

**Comment by saghul** [comment]
> not sure why this disabled style is needed. Isn't the disabled style a standard style that should be the same for every overflow menu components?

It's not a standard style. You could decide to show the text with a strike-through, I chose the opacity change, which I find more visually appealing. 

We need it to be a separate style because the component which uses the button doesn't know its state, so it doesn't know what style to pass: the disabled or non-disabled style. The idea here is that the user passes both, and the component is smart enough to choose.

> This allows us to have a different style inside the same menu, which would be bad, I suppose

That's already the case, every view can get a different style, so that point is moot.

File: react/features/base/toolbox/components/AbstractToolboxItem.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/base/toolbox/components/AbstractToolboxItem.js]: Sorry, github trimmed my comment, what I meant Array < Object >

**Comment by saghul** [comment]
> And yes, my whole point was that the component knows its own internal state (toggled or not), so it can apply the opacity change on itself automagically, without the user(developer) need to tell him what a "disabled style" means.

Again, that might be true for the overflow menu, but it's probably not what you want to do if the button is in the toolbar. Or maybe you want a different opacity. The whole point is to leave if flexible, just like the rest of the styling. It wouldn't make sense to have all the styles flexible and this one hardcoded.

File: react/features/base/toolbox/components/AbstractToolboxItem.js (hunk 1 above)
Comment (saghul) [reply react/features/base/toolbox/components/AbstractToolboxItem.js]: I couldn’t get flow to shut up :-(

**Comment by saghul** [comment]
Jenkins please test this please.

File: react/features/base/toolbox/components/AbstractToolboxItem.js (hunk 1 above)
Comment (zbettenbuk) [reply react/features/base/toolbox/components/AbstractToolboxItem.js]: I use the same type markup in the indicators now, let me spend a bit getting back on this

Comment (zbettenbuk) [reply react/features/base/toolbox/components/AbstractToolboxItem.js]: Ok, so the issue is the following: You allow the [icon|label]Style to be an Object or an Array < Object > but then in the AbstractButton you also do the following assignment:
iconStyle: [ buttonStyles.iconStyle, disabledStyles.iconStyle ]
where, if iconStyle is an Array < Object > , it ends up as an Array < Array < Object > >. And that's why it accepts Array < * > but not Array < Object >

So flow shouldn't shut up as it's warning you about a possible bug ;)

Comment (zbettenbuk) [reply react/features/base/toolbox/components/AbstractToolboxItem.js]: Btw pls take a look at this function as I was dealing with something similar recently: https://github.com/zbettenbuk/jitsi-meet/blob/fe1447b5551e6a879f0a7c0c9ffbb6a24376e5a6/react/features/base/styles/functions.js#L76

**Comment by saghul** [comment]
Cool, I didn’t know you had added that! I’ll take a look tomorrow!

**Comment by saghul** [comment]
Thanks @zbettenbuk for your patience and thoroughness, I think we arrived at a better solution than the one I initially had in mind! 👏

**Pull request merged**